specs/protocol/holocene/configurability.md

```diff
@@ -0,0 +1,136 @@
+# Configurability
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Constants](#constants)
+  - [`ConfigType`](#configtype)
+- [`SystemConfig`](#systemconfig)
+  - [`ConfigUpdate`](#configupdate)
+  - [Modifying EIP-1559 Parameters](#modifying-eip-1559-parameters)
+  - [Interface](#interface)
+    - [EIP-1559 Params](#eip-1559-params)
+      - [`setEIP1559Params`](#seteip1559params)
+    - [Fee Vault Config](#fee-vault-config)
+      - [`setBaseFeeVaultConfig`](#setbasefeevaultconfig)
+      - [`setL1FeeVaultConfig`](#setl1feevaultconfig)
+      - [`setSequencerFeeVaultConfig`](#setsequencerfeevaultconfig)
+- [`OptimismPortal`](#optimismportal)
+  - [Interface](#interface-1)
+    - [`setConfig`](#setconfig)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `SystemConfig` and `OptimismPortal` are updated with a new flow for chain
+configurability.
+
+## Constants
+
+### `ConfigType`
+
+The `ConfigType` enum represents configuration that can be modified.
+
+| Name | Value | Description |
+| ---- | ----- | --- |
+| `SET_GAS_PAYING_TOKEN` | `uint8(0)` | Modifies the gas paying token for the chain |
+| `SET_BASE_FEE_VAULT_CONFIG` | `uint8(1)` | Sets the Fee Vault Config for the `BaseFeeVault` |
+| `SET_L1_FEE_VAULT_CONFIG` | `uint8(2)` | Sets the Fee Vault Config for the `L1FeeVault` |
+| `SET_SEQUENCER_FEE_VAULT_CONFIG` | `uint8(3)` | Sets the Fee Vault Config for the `SequencerFeeVault` |
+| `SET_L1_CROSS_DOMAIN_MESSENGER_ADDRESS` | `uint8(4)` | Sets the `L1CrossDomainMessenger` address |
+| `SET_L1_ERC_721_BRIDGE_ADDRESS` | `uint8(5)` | Sets the `L1ERC721Bridge` address |
+| `SET_L1_STANDARD_BRIDGE_ADDRESS` | `uint8(6)` | Sets the `L1StandardBridge` address |
+| `SET_REMOTE_CHAIN_ID` | `uint8(7)` | Sets the chain id of the base chain |
+
+## `SystemConfig`
+
+### `ConfigUpdate`
+
+The following `ConfigUpdate` enum is defined where the `CONFIG_VERSION` is `uint256(0)`:
+
+| Name | Value | Definition | Usage |
+| ---- | ----- | --- | -- |
+| `BATCHER` | `uint8(0)` | `abi.encode(address)` | Modifies the account that is authorized to progress the safe chain |
+| `GAS_CONFIG` | `uint8(1)` | `(uint256(0x01) << 248) \| (uint256(_blobbasefeeScalar) << 32) \| _basefeeScalar` | Modifies the fee scalars |
+| `GAS_LIMIT` | `uint8(2)` | `abi.encode(uint64 _gasLimit)` | Modifies the L2 gas limit |
+| `UNSAFE_BLOCK_SIGNER` | `uint8(3)` | `abi.encode(address)` | Modifies the account that is authorized to progress the unsafe chain |
+| `EIP_1559_PARAMS` | `uint8(4)` | `abi.encode(uint64 _denominator, uint64 _elasticity)` | Modifies the EIP-1559 denominator and elasticity |
+
```

Comment:
We could add a section on `Initialization` that clearly defines which calls should be made during an initialize call